packages/wrangler/src/pages.tsx

```diff
@@ -821,25 +821,31 @@ export const pages: BuilderCallback<unknown, unknown> = (yargs) => {
         ...miniflareArgs,
       });
 
-      const server = await miniflare.startServer();
-      console.log(`Serving at http://127.0.0.1:${port}/`);
+      try {
+        // `startServer` might throw if user code contains errors
+        const server = await miniflare.startServer();
+        console.log(`Serving at http://127.0.0.1:${port}/`);
 
-      if (process.env.BROWSER !== "none") {
-        await open(`http://127.0.0.1:${port}/`);
-      }
+        if (process.env.BROWSER !== "none") {
+          await open(`http://127.0.0.1:${port}/`);
+        }
 
-      process.on("SIGINT", () => {
-        server.close();
-        miniflare.dispose().catch((err) => {
-          console.error(err);
+        process.on("SIGINT", () => {
+          server.close();
+          miniflare.dispose().catch((err) => {
+            miniflare.log.error(err);
+          });
         });
-      });
-      process.on("SIGTERM", () => {
-        server.close();
-        miniflare.dispose().catch((err) => {
-          console.error(err);
+        process.on("SIGTERM", () => {
+          server.close();
+          miniflare.dispose().catch((err) => {
+            miniflare.log.error(err);
+          });
         });
-      });
+      } catch (e) {
+        miniflare.log.error(e);
+        process.exitCode = 1;
```

Comment:
This exit doesn't quite do enough to kill the process, but it's not blocking and likely will change with my #160 anyway.